torch/nn/modules/module.py

```diff
@@ -2021,6 +2043,7 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
 
         def load(module, local_state_dict, prefix=''):
             local_metadata = {} if metadata is None else metadata.get(prefix[:-1], {})
+            local_metadata['assign_to_params_buffers'] = int(assign)
```

Comment:
This might be a very debatable design choice here to put the flag into the `local_metadata` dict.

I chose to do this instead of adding a new `assign` argument to `_load_from_state_dict` because it seems like `_load_from_state_dict` is overridden **a lot** in practice. 

So if I had 
```
module._load_from_state_dict(
                local_state_dict, prefix, local_metadata, True, missing_keys, unexpected_keys, error_msgs, assign)
```
below it would break with every module that overrides `_load_from_state_dict`. 

We could fix all the modules we own (and a ton more after ghimport) but it would still break a lot of external modules so I'm very hesitant to change the signature of `_load_from_state_dict`... 

Wondering whether there is a better way around this that I'm missing here?

Reply:
That sounds like an ok workaround for me